src/Functions/CastOverloadResolver.h

```diff
@@ -28,6 +28,6 @@ struct CastDiagnostic
     std::string column_to;
 };
 
-FunctionBasePtr createInternalCast(ColumnWithTypeAndName from, DataTypePtr to, CastType cast_type, std::optional<CastDiagnostic> diagnostic);
+FunctionBasePtr createInternalCast(ColumnWithTypeAndName from, DataTypePtr to, CastType cast_type, std::optional<CastDiagnostic> diagnostic, ContextPtr context = nullptr);
```

Comment:
We should not use `nullptr` as default argument here. We just need to always propagate context on each call of `createInternalCast`

Reply:
I think I got all the spots now where I could propagate the contaxt towards internal cast, at some points I couldn't find a way to get a context without changing a lot of code.